CMakeLists.txt

```diff
@@ -62,8 +62,14 @@ set_target_properties(aws-checksums PROPERTIES LINKER_LANGUAGE C)
 set(CMAKE_C_FLAGS_DEBUGOPT "")
 set_property(TARGET aws-checksums PROPERTY C_STANDARD 99)
 
-if(BUILD_SHARED_LIBS AND WIN32)
-    target_compile_definitions(aws-checksums PRIVATE "-DAWS_CHECKSUMS_EXPORTS")    
+if(BUILD_SHARED_LIBS)
+    if (WIN32)
+        target_compile_definitions(aws-checksums PRIVATE "-DAWS_CHECKSUMS_EXPORTS")    
+    endif()
+    set_target_properties(aws-checksums
+        PROPERTIES
+            VERSION ${PROJECT_VERSION}
+            SOVERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR})
```

Comment:
The AWS C++ SDK seems to rely on the unversioned `.so`, so I think I need to remove this.